### ISSUE TITLE
Job handlers are not running in parallel

### ISSUE THREAD

**Issue by MrRickwi** [zeebe-io/zeebe-client-csharp#157]
**Describe the bug**
When implementing a simple job handler that blocks for a certain amount of time it can bee seen that only one worker at a time is running, ignoring the property `MaxJobsActive`. This is due to the fact that a queue is used for running the workers one after each other.

**To Reproduce**
- Create a simple workflow that has one activity "sleep".
- Register a worker in your C# client with a simple `Thread.Sleep(TimeSpan.FromSeconds(60))` inside
- Start more than `MaxJobsActive` instances of the sleep workflow
- In SimpleMonitor it can be seen that only one job at a time is consumed

**Expected behavior**
`MaxJobsActive` jobs should be run in parallel. When the first bunch of jobs is finished the next one should start.

**Enviroment (please complete the following information):**
 - OS: Windows 10
 - Version 0.24.1 of Zeebe, 0.16.1 of C# client

**Additional context**
![Sleep](https://user-images.githubusercontent.com/66736825/89638679-8379e600-d8ac-11ea-96b7-6b1f5424d2aa.png)

**Comment by MrRickwi** [comment]
I saw your branch zell-rewrite-job-worker. Tested it. It solves the problem somehow, but in my opinion not correctly. `MaxJobsActive` is now duplicated by `HandlerThreads`, so if you have `MaxJobsActive = 10` and `HandlerThreads = 3`, my client activates 10 jobs, but handles only 3 in parallel. I could supply a much easier solution in the current code if desired.

**Comment by Zelldon** [comment]
Hey @MrRickwi, 

thanks for reporting this. This is actually a duplicate of https://github.com/zeebe-io/zeebe-client-csharp/issues/133.
This issue is a missing feature, not a bug and as you already saw I'm currently working on it. I hope I get a release out soon.

The `maxJobsActive` has a different concern then the `HandlerThreads`, since it is used to pre-fetch jobs from the broker. You can also take a look in the other clients how it is implemented there. More information you can also find in the docs https://docs.zeebe.io/reference/grpc.html#activatejobs-rpc

**Comment by MrRickwi** [comment]
Thanks for your comment. Possibly it is a good idea to initialize `HandlerThreads` with `MaxJobsActive` for a better backwards-compatibility? Makes sense to close my pull request if yours is handling the situation better.
